assignment/models/homework.py

```diff
@@ -48,6 +48,19 @@ def check_date(self):
     def onchange_subject_standard(self):
         self.subject_standard_assignment = self.standard_id.standard_id.id
 
+    @api.onchange('teacher_id')
+    def onchange_teacher_id(self):
+        self.standard_id = False
+        if self.teacher_id:
+            domain = [(('teacher_ids', 'in', [self.teacher_id.id]))]
+            return {'domain': {'subject_id': domain}}
```

Comment:
Did you review @AnuKPatel ?

Reply:
I have tested the it functionally, but seems like the code is not up to the mark.
Let me ask Nisarg to improve it.